botcore/async_stats.py

```diff
@@ -0,0 +1,59 @@
+"""An async transport method for statsd communication."""
+
+import asyncio
+import socket
+from typing import Optional
+
+from discord.utils import MISSING
+
+from botcore.utils import scheduling
+
+try:
+    from statsd.client.base import StatsClientBase
+except ImportError:
+    StatsClientBase = MISSING
```

Comment:
It would be nicer to make `AsyncStatsClient` `MISSING` but I'm not sure of a nice way to do that.

Reply:
Possibly after the definition we check if `statsd` is loaded, if not we redefine it as `MISSING`?

Reply:
Yeah, that sounds good. It's nice to be able to consistently check using `x is Missing` rather than worrying about whether `isinstance` has to be used sometimes.

Reply:
Please document somewhere that redis and statsd are optional as well as how to check if something is missing. A stray thought is whether an import error should be raised if imports are attempted (except from within botcore).

Reply:
https://github.com/python-discord/bot-core/pull/42/commits/49df21e48da18c01c6bea1cbb9ce1187e89adcf3